kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/hover/KotlinJavadocTextHover.kt

```diff
@@ -16,52 +16,198 @@
+import org.eclipse.jface.text.DefaultInformationControl
+import org.eclipse.jface.text.IInformationControl
 import org.eclipse.jface.text.IInformationControlCreator
-import org.eclipse.jface.text.Region
-import org.jetbrains.kotlin.core.model.toJavaElements
-import org.jetbrains.kotlin.core.references.createReferences
+import org.eclipse.jface.text.ITextHoverExtension
+import org.eclipse.swt.graphics.RGB
+import org.eclipse.swt.widgets.Shell
+import org.jetbrains.kotlin.core.model.KotlinAnalysisFileCache
 import org.jetbrains.kotlin.core.references.getReferenceExpression
-import org.jetbrains.kotlin.core.references.resolveToSourceElements
+import org.jetbrains.kotlin.core.references.resolveToSourceDeclaration
 import org.jetbrains.kotlin.core.resolve.lang.java.resolver.EclipseJavaSourceElement
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.ClassKind.ANNOTATION_CLASS
+import org.jetbrains.kotlin.descriptors.ClassKind.CLASS
+import org.jetbrains.kotlin.descriptors.ClassKind.ENUM_CLASS
+import org.jetbrains.kotlin.descriptors.ClassKind.ENUM_ENTRY
+import org.jetbrains.kotlin.descriptors.ClassKind.INTERFACE
+import org.jetbrains.kotlin.descriptors.ClassKind.OBJECT
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptor
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptorWithVisibility
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.descriptors.ValueDescriptor
+import org.jetbrains.kotlin.idea.kdoc.findKDoc
+import org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageFragment
+import org.jetbrains.kotlin.psi.KtDeclaration
 import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.resolve.BindingContext.REFERENCE_TARGET
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.jetbrains.kotlin.resolve.descriptorUtil.isExtension
+import org.jetbrains.kotlin.resolve.descriptorUtil.isExtensionProperty
+import org.jetbrains.kotlin.resolve.lazy.descriptors.LazyClassDescriptor
 import org.jetbrains.kotlin.ui.editors.KotlinEditor
+import org.jetbrains.kotlin.ui.editors.navigation.findDeclarationInParsedFile
+import org.jetbrains.kotlin.ui.editors.navigation.getKtFileFromElement
+import org.jetbrains.kotlin.ui.editors.quickassist.resolveToDescriptor
+import java.net.URL
 
-class KotlinJavadocTextHover : KotlinEditorTextHover<BrowserInformationControlInput> {
+class KotlinJavadocTextHover : KotlinEditorTextHover<BrowserInformationControlInput>, ITextHoverExtension {
+
+    class KotlinInformationControlCreator : IInformationControlCreator {
+
+        override fun createInformationControl(parent: Shell?): IInformationControl =
+            if (BrowserInformationControl.isAvailable(parent)) {
+                val tbm = ToolBarManager(8388608)
+                val font = "org.eclipse.jdt.ui.javadocfont"
+                BrowserInformationControl(parent, font, tbm)
```

Comment:
`tbm` and `font` are constants, so let's extract them to a `companion` object, for example. Plus, please explain why `8388608` is used